keras2onnx/_builtin.py

```diff
@@ -140,6 +141,94 @@ def convert_tf_gather_v2(scope, operator, container):
                               axis=axis)
 
 
+def _make_range_const(scope, operator, container, start, limit, delta, onnx_type):
+    start = _cal_tensor_value(start).tolist()
+    limit = _cal_tensor_value(limit).tolist()
+    delta = _cal_tensor_value(delta).tolist()
+    val = np.arange(start, limit, delta)
+    oopb = OnnxOperatorBuilder(container, scope)
+    oopb.add_node_with_output('Identity',
+                             [('_start', onnx_type, val)],
+                             operator.outputs[0].full_name,
+                             name=operator.full_name + '_range')
+
+
+def _make_range_non_const(scope, operator, container, start, limit, delta, onnx_type):
+    oopb = OnnxOperatorBuilder(container, scope)
+    diff_node = oopb.apply_sub([limit.name, start.name],
+                               name=operator.full_name + '_diff')
+    delta_cast = delta.name
+    if onnx_type in [onnx_proto.TensorProto.INT32, onnx_proto.TensorProto.INT64]:
+        diff_output = oopb.apply_cast(diff_node,
+                                    to=onnx_proto.TensorProto.FLOAT,
+                                    name=operator.full_name + '_cast_diff')
+        delta_cast = oopb.apply_cast(delta.name,
+                                     to=onnx_proto.TensorProto.FLOAT,
+                                     name=operator.full_name + '_cast_delta')
+
+    div_node = oopb.apply_div(diff_output + delta_cast,
+                               name=operator.full_name + '_div')
+    ceil_node = oopb.add_node("Ceil",
+                               div_node,
+                               name=operator.full_name + '_ceil')
+    trip_count_node = oopb.apply_cast(ceil_node,
+                                      to=onnx_proto.TensorProto.INT64,
+                                      name=operator.full_name + '_trip_cnt')
+    loop_inputs = [trip_count_node[0],
+                   # TENSOR_TYPE_TO_STORAGE_TENSOR_TYPE maps BOOL to INT32
+                   # so we need change np.array(True, dtype='bool') to int32 here
+                   ('_cond', onnx_proto.TensorProto.BOOL, np.array(1, dtype='int32')),
+                   start.name]
+    from onnx import helper
+    n1 = helper.make_node("Identity", ["cond"], ["cond_out"], name="n1")
```

Comment:
should this be extracted as a function?

Reply:
Since no other places use this structure, let's keep it for now until needed.